internal/app/machined/pkg/controllers/runtime/extension_services_config.go

```diff
@@ -0,0 +1,93 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package runtime
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/safe"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/siderolabs/gen/optional"
+	"go.uber.org/zap"
+
+	"github.com/siderolabs/talos/pkg/machinery/resources/config"
+	"github.com/siderolabs/talos/pkg/machinery/resources/runtime"
+)
+
+// ExtensionServicesConfigController watches v1alpha1.Config, creates/updates/deletes extension services config.
+type ExtensionServicesConfigController struct{}
+
+// Name implements controller.Controller interface.
+func (ctrl *ExtensionServicesConfigController) Name() string {
+	return "runtime.ExtensionServicesConfigController"
+}
+
+// Inputs implements controller.Controller interface.
+func (ctrl *ExtensionServicesConfigController) Inputs() []controller.Input {
+	return []controller.Input{
+		{
+			Namespace: config.NamespaceName,
+			Type:      config.MachineConfigType,
+			ID:        optional.Some(config.V1Alpha1ID),
+		},
+	}
```

Comment:
if we want to track and properly destroy ouputs with finalizers, we need to add outputs as an input with `Kind: InputDestroyReady`